superbench/benchmarks/model_benchmarks/pytorch_base.py

```diff
@@ -189,6 +189,36 @@ def _create_optimizer(self):
 
         return True
 
+    def _is_finished(self, curr_step, curr_time, check_frequency=100):
+        """Judge whether the benchmarking should be stopped early or not.
+
+        Args:
+            curr_step (int): the current benchmarking step.
+            curr_time (float): the current time in seconds got from time.time().
+            check_frequency (int): the frequency (step numbers) to check if benchmark should be stopped.
+
+        Return:
+            True if the benchmarking should be stopped.
+        """
+        is_finished = super()._is_finished(curr_step, curr_time)
+        is_finished = 1 if is_finished is True else 0
+        if self._args.duration > 0:
+            if curr_step % check_frequency == 0:
+                # sync is_finished in distributed mode
+                # if any rank is_finished is True, all ranks should be finished
+                if self._args.distributed_impl == DistributedImpl.DDP:
+                    result = [is_finished]
+                    if self._args.distributed_backend == DistributedBackend.NCCL:
+                        tensor = torch.as_tensor(result).cuda()
+                    else:
+                        tensor = torch.as_tensor(result)
```

Comment:
```suggestion
                    tensor = torch.ones([is_finished], dtype=torch.int32)
                    if self._args.distributed_backend == DistributedBackend.NCCL:
                        tensor = tensor.cuda()
```

Reply:
torch.ones? it seems wrong to me. https://pytorch.org/docs/stable/generated/torch.ones.html#torch-ones

Reply:
updated..

Reply:
updated, thanks